File: src/dataset/go/rna_transcription_Exercism_go_rootulp/claude/ClaudeSonnet4_rna_transcription_v2.go
package strand

func ToRNA(dna string) string {
	if len(dna) == 0 {
		return ""
	}
	
	result := make([]byte, len(dna))
	for i := 0; i < len(dna); i++ {
		switch dna[i] {
		case 'C':
			result[i] = 'G'
		case 'G':
			result[i] = 'C'
		case 'T':
			result[i] = 'A'
		case 'A':
			result[i] = 'U'
		}
	}
	return string(result)
}

File: src/dataset/javascript/rna-transcription_exercism-javascript-ThomasZumsteg/claude/ClaudeSonnet4_rna-transcription_v2.js
var DnaTranscriber = function() {};

// Pre-computed mapping to avoid recreation on each call
const DNA_TO_RNA_MAP = { 'G': 'C', 'C': 'G', 'T': 'A', 'A': 'U' };

DnaTranscriber.prototype.toRna = function(dna) {
	var result = '';
	for (var i = 0; i < dna.length; i++) {
		result += DNA_TO_RNA_MAP[dna[i]];
	}
	return result;
}

export default DnaTranscriber;